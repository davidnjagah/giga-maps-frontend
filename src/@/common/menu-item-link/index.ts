export { default } from './menu-item-link.view'