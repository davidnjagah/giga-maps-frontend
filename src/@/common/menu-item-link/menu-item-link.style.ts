import { ContainedListItem, SideNavLink } from '@carbon/react';
import styled, { css } from 'styled-components';


export const MenuItemBlank = styled(SideNavLink)`
display: flex;
align-items: center;
padding: 0.8rem 1rem 0.8rem 1rem !important;
cursor: pointer; 
color: ${props => props.theme.main};
&:hover {
   background: ${props => props.theme.text} !important;
  .cds--side-nav__link-text {
    color:  ${props => props.theme.main} !important;
    font-weight: normal;
  }
}

>span.cds--side-nav__link-text {
  color:  ${props => props.theme.text};
  font-weight: normal;
}
>.cds--side-nav__icon {
  margin-right: 0.7rem;
}
.cds--side-nav__icon svg{
  fill: ${props => props.theme.text};
}
&:hover .cds--side-nav__icon svg {
  fill: ${props => props.theme.main};
}
`

export const MenuItem = styled(ContainedListItem) <{ $active?: boolean }>`
  .cds--contained-list-item__content:not(:disabled):hover {
    color: ${props => props.theme.main};
    background:${props => props.theme.text};
  }

.cds--contained-list-item__content {
display: flex;
align-items: center;
padding: 0.8rem 1rem 0.8rem 1rem !important;
cursor: pointer;
${(props) => props.$active && css`
  color: ${props => props.theme.main} !important;
  background:${props => props.theme.text};
`}
}
>.cds--contained-list-item__icon, >.cds--side-nav__icon {
  margin-right: 0.7rem;
}

>.cds--contained-list-item__content, >.cds--side-nav__link {
  color: ${props => props.theme.text};
  font-family: Open Sans;
  font-size: 0.875rem;
  font-style: normal;
  font-weight: 400;
  line-height: 1.125rem;
  
}
`

export const SideBarMenuList = styled.div<{ $style?: ReturnType<typeof css> }>`
background: ${props => props.theme.main};
height: calc(100% - 3rem);
top:3rem;
width:100%;
    position: absolute;
    z-index: 6;
.cds--contained-list-item:not(:last-of-type)::before {
  background-color: ${props => props.theme.main}  !important;
}

@media (max-width:768px){
  position: fixed;
}
${props => props.$style}
`
export const MobileFooterContainer = styled.div`
position: fixed;
    bottom: 0;
    padding: 1rem;
    display: flex;
    align-items: center;

     .link {
    font-size: 0.75rem;
    margin-left: 0.75rem;
    text-transform: none;
  }

  a {
    margin-left: .5rem;
  }

  p{
    font-size: 0.75rem;
    color: ${props => props.theme.text};
  }
  svg{
    fill: ${props => props.theme.text};
  }
`
