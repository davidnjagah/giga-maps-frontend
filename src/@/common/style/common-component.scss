.side-info-download-api-signup {
  padding: 1.5rem 1rem 1rem 1rem;

  >button {
    width: 100%;
  }
}

.side-info-download-api-avatar-username {
  display: flex;
  align-items: center;
}

.side-info-download-api-avatar {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.375rem;
  height: 2.375rem;
  border-radius: 50%;
  background: #277AFF;
  color: #fff;
}

.side-info-download-api-username {
  margin-left: 0.5rem;

  >span {
    color: var(--text-text-primary, #222);
    font-size: 1rem;
    font-style: normal;
    font-weight: 500;
    line-height: 1.375rem;
  }

  >p {
    color: var(--text-text-tertiary, #6F6F6F);
    font-size: 0.75rem;
    font-style: normal;
    font-weight: 400;
    line-height: 1rem;
    letter-spacing: 0.02rem;
  }
}

.side-info-download-api-bottom-line {
  width: 100%;
  height: 0.0625rem;
  background: #E6E6E6;
  margin-top: 0.8rem
}