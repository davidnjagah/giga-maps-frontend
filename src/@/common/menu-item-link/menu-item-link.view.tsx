import { PropsWithChildren } from 'react';

import { setTheme } from '~/core/theme.model';
import { Link } from '~/lib/router';
import { Route } from '~/lib/router/types';

import { MenuItem } from './menu-item-link.style';

const MenuItemLink = ({ icon, children, to, isActive, id }: PropsWithChildren<{ icon: any; to: Route, isActive?: boolean; id?: string }>) => {

  return (
    <Link to={to} id={id}>
      <MenuItem renderIcon={icon} $active={isActive}>
        {children}
      </MenuItem>
    </Link>
  )
}

export default MenuItemLink