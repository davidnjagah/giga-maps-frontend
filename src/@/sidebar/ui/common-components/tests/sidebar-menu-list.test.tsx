import { fireEvent, render } from '@testing-library/react';
import { testWrapper } from '~/tests/jest-wrapper';
import SidebarMenuList from '../sidebar-menu-list';
import { MenuItemBlank } from '~/@/common/menu-item-link/menu-item-link.style';
import { Api } from '@carbon/icons-react';

describe('Sidebar menu list component', () => {
  it('should render component', () => {
    const { asFragment } = render(testWrapper(<SidebarMenuList />));
    expect(asFragment).toMatchSnapshot();
  });

  it('should render component and handle click', () => {
    const onChangeMenu = jest.fn();
    const { getByText } = render(
      <MenuItemBlank onClick={() => onChangeMenu(false)} href={'/docs/explore-api'} renderIcon={Api} target="_blank">
        API’s and Download
      </MenuItemBlank>
    );

    fireEvent.click(getByText("API’s and Download"));

    expect(onChangeMenu).toHaveBeenCalledTimes(1);

    expect(getByText("API’s and Download")).toBeInTheDocument();
  });
});
