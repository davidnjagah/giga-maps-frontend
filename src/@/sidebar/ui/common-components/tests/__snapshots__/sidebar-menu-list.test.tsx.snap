// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar menu list component should render component 1`] = `[Function]`;
